String.php

```diff
@@ -755,32 +790,59 @@ public function getWidth ( ) {
     public function getDirection ( ) {
 
         if(null === $this->_direction) {
-
-            if(null === $this->_string)
-                $this->_direction = static::LTR;
-            else
-                $this->_direction = static::getCharDirection(
-                    mb_substr($this->_string, 0, 1)
-                );
+            // Default
+            $this->_direction = static::LTR;
+
+            // Check for LRM or RLM/ARM
+            $hasLRM      = 0 !== preg_match('#\x{200e}#u', $this->_string);
+            $hasRLMOrARM = 0 !== preg_match('#\x{200f}|\x{061c}#u', $this->_string);
+
+            if($hasLRM || $hasRLMOrARM) {
```

Comment:
Can you explain me (in details) the following algorithm please :-)? It will ease the review. Thanks!